Compiling for X11 aarch64 fails (undefined reference linking)

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
master


**OS/device including version**
Termux Arch on Android phone


**Issue description:**
Compiling for the aarch64 doesn't work natively in a termux Emulator with arch repo.
I've previously disabled yasm (with yasm there are other issues).

Command:
```scons p=x11 target=release_debug tools=yes arch=aarch64 -j 1```

**Error Output:**
```
/Recast/Source/RecastRegion.cpp              [Initial build] Linking Static Library ==> modules/recast/librecast_builtin.x11.opt.tools.aarch64.a   Ranlib Library         ==> modules/recast/librecast_builtin.x11.opt.tools.aarch64.a                   [Initial build] Linking Program        ==> bin/godot.x11.opt.tools.aarch64                            modules/libmodules.x11.opt.tools.aarch64.a(systemdependent.x11.opt.tools.aarch64.o): In function `vp8_machine_specific_config':
/godot/thirdparty/libvpx/vp8/common/generic/systemdependent.c:102: undefined reference to `arm_cpu_caps'                                                 collect2: error: ld returned 1 exit status
scons: *** [bin/godot.x11.opt.tools.aarch64] Error 1
scons: building terminated because of errors.      [19:06 godot ]$

```

**Minimal reproduction project:**
<!-- Recommended as it greatly speeds up debugging. Drag and drop a zip archive to upload it. -->

--- a/drivers/unix/os_unix.cpp
+++ b/drivers/unix/os_unix.cpp
@@ -244,7 +244,7 @@ OS::TimeZoneInfo OS_Unix::get_time_zone_info() const {
 
 void OS_Unix::delay_usec(uint32_t p_usec) const {
 
-	struct timespec rem = { p_usec / 1000000, (p_usec % 1000000) * 1000 };
+	struct timespec rem = { static_cast<time_t>(p_usec / 1000000), static_cast<long>((p_usec % 1000000) * 1000) };
 	while (nanosleep(&rem, &rem) == EINTR) {
 	}
 }

--- a/modules/webm/libvpx/SCsub
+++ b/modules/webm/libvpx/SCsub
@@ -263,7 +263,7 @@ if env["platform"] == 'uwp':
         webm_cpu_x86 = True
 else:
     import platform
-    is_x11_or_server_arm = ((env["platform"] == 'x11' or env["platform"] == 'server') and platform.machine().startswith('arm'))
+    is_x11_or_server_arm = ((env["platform"] == 'x11' or env["platform"] == 'server') and (platform.machine().startswith('arm') or platform.machine().startswith('aarch')))
     is_ios_x86 = (env["platform"] == 'iphone' and env["ios_sim"])
     is_android_x86 = (env["platform"] == 'android' and env["android_arch"] == 'x86')
     if is_android_x86:
@@ -337,7 +337,6 @@ if webm_cpu_arm:
 if webm_simd_optimizations == False:
     print("WebM SIMD optimizations are disabled. Check if your CPU architecture, CPU bits or platform are supported!")
 
-
 env_libvpx.add_source_files(env.modules_sources, libvpx_sources)
 if webm_cpu_x86:
     is_clang_or_gcc = ('gcc' in env["CC"]) or ('clang' in env["CC"]) or ("OSXCROSS_ROOT" in os.environ)
@@ -379,5 +378,5 @@ elif webm_cpu_arm:
         env_libvpx.add_source_files(env.modules_sources, libvpx_sources_arm_neon_armasm_ms)
     elif env["platform"] == 'iphone':
         env_libvpx.add_source_files(env.modules_sources, libvpx_sources_arm_neon_gas_apple)
-    elif not env["android_arch"] == 'arm64v8':
+    elif (is_x11_or_server_arm and cpu_bits == '32') or (env["platform"] == 'android' and not env["android_arch"] == 'arm64v8'):
         env_libvpx.add_source_files(env.modules_sources, libvpx_sources_arm_neon_gas)

